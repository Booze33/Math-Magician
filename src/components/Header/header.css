.header {
  width: 100vw;
  height: 4rem;
  display: flex;
  justify-content: space-around;
  align-items: center;
  margin-top: -8rem;
  border-bottom: 2px solid #000;
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
  border-bottom-left-radius: 58px;
  background: #5e1ca0;
  box-shadow: inset -22px -12px 44px #310760, inset 22px 22px 44px #d4a8ff;
}

.h1 {
  font-size: 40px;
  font-weight: 900;
  font-style: italic;
  animation: header 1s ease 1;
  margin-left: -2rem;
}

@keyframes header {
  0% {
    transform: translateX(100rem);
  }

  50% {
    transform: translateX(100rem);
  }

  100% {
    transform: translateX(0);
  }
}

.button {
  margin: 0 0.5rem 0 0.5rem;
  padding: 7px;
  height: 2rem;
  border: none;
  color: #2b2234;
  text-decoration: none;
  font-size: 1.5rem;
  transition: all 1.5s;
  font-style: italic;
  border-radius: 71px;
  background: #5e1ca0;
  box-shadow: inset 13px 13px 26px #431472, inset -13px -13px 26px #7924ce;
}

.button:hover {
  text-decoration: underline;
  color: #000;
  font-size: 1.8rem;
  font-style: oblique;
  transition: all 1.5s;
}

button {
  margin: 0 0.3rem 0 0.5rem;
  height: 2rem;
  border: none;
  background-color: transparent;
}

nav {
  margin-left: 3rem;
}
