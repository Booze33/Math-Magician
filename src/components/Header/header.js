import React from 'react';
import { Outlet, Link } from 'react-router-dom';
import './header.css';

function Header() {
  return (
    <div className="header">
      <h1 className="h1">Math Magic</h1>
      <nav className="nav">
        <button type="button">
          <Link to="/" className="button">Home</Link>
        </button>
        <button type="button">
          <Link to="/calculator" className="button">Calculator</Link>
        </button>
        <button type="button">
          <Link to="/quote" className="button">Quote</Link>
        </button>
      </nav>
      <Outlet />
    </div>
  );
}

export default Header;
