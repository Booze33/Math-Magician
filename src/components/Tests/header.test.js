import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from '../Header/header';

afterEach(() => {
  cleanup();
});

test('Test loading screen', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
