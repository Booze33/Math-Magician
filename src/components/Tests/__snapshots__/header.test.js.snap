// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <div
    class="header"
  >
    <h1
      class="h1"
    >
      Math Magic
    </h1>
    <nav
      class="nav"
    >
      <button
        type="button"
      >
        <a
          class="button"
          href="/"
        >
          Home
        </a>
      </button>
      <button
        type="button"
      >
        <a
          class="button"
          href="/calculator"
        >
          Calculator
        </a>
      </button>
      <button
        type="button"
      >
        <a
          class="button"
          href="/quote"
        >
          Quote
        </a>
      </button>
    </nav>
  </div>
</div>
`;
